Комулов Андрей

Andreyproger opened the pull request.

Andreyproger committed:
Комулов Андрей

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,64 @@
+ * @param {array} array
+ * @returns {array} key is a unique element of this array, 
+ *  value is a number of occurrences
+ **/
```
Я не понял к чему относится этот комментарий.
Он должен идти прямо перед функцией, которую описывает

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,64 @@
+
+
 function getPokerHand(dice) {
+	// use Энтропия Шеннона
```
Что?

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,64 @@
+    	throw new Error("It isn't Array!");
+    }
+
+    if(dice.some(j => isNaN(j) || j <= 0 || j >= 7)){
```
Элемент массива может быть числом в диапазоне от 1 до 6, но все равно не подходящий под задачу. Например, `4.5`

Zhigalov replied:
Для справки
```js
isNaN([1]); // false
```

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,64 @@
+    	throw Error("Length of Array not equal 5!");
+    }
+
+    const gFM = getFrequencyMap(dice)
```
`gFM`? В JavaScript не принято давать такие названия переменным.
Я бы назвал `frequencyMap`

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,64 @@
+
+    const gFM = getFrequencyMap(dice)
+    	.filter(j => j); //getFrequencyMap
+    const counts = gFM
```
Эта переменная нигде не используется

Zhigalov commented on tests/getPokerHand-test.js:
```diff
@@ -1,12 +1,89 @@
+	    it('should return `Тройка` for [1, 1, 1, 4, 3]', () =>{
+	    	const actual = getPokerHand([1, 1, 1, 4, 3]);
+
+	    	assert.equal(actual, `Пара`);
```
Очень много тестов для "Пары". В чем разница? Они не увеличивают покрытие кода

Zhigalov commented on tests/getPokerHand-test.js:
```diff
@@ -1,12 +1,89 @@
+
+            assert.equal(actual, 'Наивысшее очко');
+        });
+	});
```
Где проверка на тройку?

Zhigalov commented on tests/getPokerHand-test.js:
```diff
@@ -1,12 +1,89 @@
+	    });
+
+	    it('should return `It is not array` for [1]', () => {
+	        const actual = () => getPokerHand([]);
```
Описание теста не соответствует проверке

Zhigalov commented on tests/getPokerHand-test.js:
```diff
@@ -1,12 +1,89 @@
+
+	describe('Not Positive', () =>{
+		it('should return `It is not array` for []', () => {
+	        const actual = () => getPokerHand([]);
```
Но `[]` это же массив!

Zhigalov reviewed (changes requested):
При попытке запустить код получаю ошибку
```
/home/travis/build/urfu-2017/autotests-task-01/lib/getPokerHand.js:22
			answer[item] = 0:
			                ^
SyntaxError: Unexpected token :
```

Zhigalov commented:
Жду исправлений

Andreyproger commented:
Исправления смогу внести только завтра.12.04.2018, 19:57, "Sergey Zhigalov" <notifications@github.com>:Жду исправлений

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  


 --------------------С уважением, Андрей Комулов

Andreyproger committed:
Update getPokerHand-test.js

Andreyproger committed:
Update getPokerHand.js

Andreyproger committed:
Update getPokerHand.js

Andreyproger commented:
Перезалил код на github через браузер, т.к. через командную строку не вышло. Можно так проверить? Если нельзя, то вышли, по возможности, команду для перезаписи через Git. 13.04.2018, 23:38, "Angus Young" <crashc@yandex.ru>:Исправления смогу внести только завтра. 12.04.2018, 19:57, "Sergey Zhigalov" <notifications@github.com>:Жду исправлений—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.    --------------------С уважением, Андрей Комулов    --------------------С уважением, Андрей Комулов 

Andreyproger committed:
Комулов Андрей

Andreyproger committed:
Merge branch 'master' of https://github.com/Andreyproger/autotests-task-01

Andreyproger committed:
Комулов Андрей

Zhigalov commented:
Твой код не собирается. Ошибка:
```
/home/travis/build/urfu-2017/autotests-task-01/tests/getPokerHand-test.js:53
		it('should return `Empty input` for ' '', () => {
		   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: missing ) after argument list
```
Подробности можешь найти если кликнешь на красный крестик у последнего коммита:
![image](https://user-images.githubusercontent.com/1654243/39058225-a8b00a76-44d4-11e8-8f42-b0a07c574bd1.png)

Andreyproger committed:
Update getPokerHand-test.js

Andreyproger committed:
Update getPokerHand.js

Change in string "dice.some(j => isNaN(j) || j <= 1 || j >= 6)" on "dice.some(j => isNaN(j) || j < 1 || j > 6)"

Andreyproger committed:
Update getPokerHand.js

Change name of function.

Andreyproger committed:
Update getPokerHand.js

Change name of array

Andreyproger committed:
Update file

Change from "Фулл хаус" to "Фулл Хаус".

Andreyproger committed:
Update getPokerHand-test.js

change fields

Andreyproger committed:
Update getPokerHand-test.js

change input in test

Andreyproger committed:
Update getPokerHand-test.js

Changes all strings in tests

Andreyproger committed:
Update getPokerHand-test.js

Andreyproger committed:
Update getPolerHand.js

Rewrite all file

Andreyproger committed:
Update getPokerHand-test.js

Andreyproger committed:
Update getPokerHand.js

rewrite file

Andreyproger committed:
Update file

Rewrite all file

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,57 @@
+        throw new TypeError('You should enter Array');
+    }
+    if (dice.length !== 5) {
+        throw new TypeError('Need throw 6 dices');
```
В игре пять костей, а не шесть

Zhigalov commented on lib/getPokerHand.js:
```diff
@@ -1,13 +1,57 @@
+    }
+    for (i=0 ; i < dice.length; i++) {
+        if (!isInteger(dice[i])) {
+            throw new TypeError('Значения должны быть целые');
```
Почему у тебя некоторые сообщения на русском, а некоторые на английском?

Zhigalov commented on tests/getPokerHand-test.js:
```diff
@@ -1,12 +1,89 @@
+
+    });
+
+    describe('negative2', () => {    
```
Почему эти два теста лежат в отдельном `describe`? 
Чем они примечательны? Можно ли их опустить на уровень "негативных тестов"?

Zhigalov commented on tests/getPokerHand-test.js:
```diff
@@ -1,12 +1,89 @@
+        }
+    });
+
+});
```
Я обычно оформляю тесты так:
```js
describe(`getPokerHand `, () => {
    describe('positive', () => {
        // Здесь собираю тесты на позитивные сценарии
    });
    describe('positive', () => {
        // Здесь собираю негативные тесты
    });
});
```

Zhigalov replied:
Старайся сделать так, чтобы в файле был только один `describe`, всё остальное вкладывай в него.

Zhigalov reviewed (commented):
Очень хорошо, осталось совсем чуть-чуть!

Andreyproger committed:
Update getPokerHand.js

Changes strings

Andreyproger committed:
Update getPokerHand-test.js

Change structure file.

Andreyproger commented:
 Так нельзя зачесть?06.05.2018, 20:28, "Sergey Zhigalov" <notifications@github.com>:@Zhigalov commented on this pull request.Очень хорошо, осталось совсем чуть-чуть!In lib/getPokerHand.js:> -    return 'Покер';
-}
-
-module.exports = getPokerHand;
+/**
+ * Определение комбинации в броске
+ *
+ * @param {Array} dice пять костей, для которых нужно определить комбинацию
+ * @returns {String} название комбинации
+ */
+function getPokerHand(dice) {
+    if (!Array.isArray(dice)) {
+        throw new TypeError('You should enter Array');
+    }
+    if (dice.length !== 5) {
+        throw new TypeError('Need throw 6 dices');
В игре пять костей, а не шестьIn lib/getPokerHand.js:> +/**
+ * Определение комбинации в броске
+ *
+ * @param {Array} dice пять костей, для которых нужно определить комбинацию
+ * @returns {String} название комбинации
+ */
+function getPokerHand(dice) {
+    if (!Array.isArray(dice)) {
+        throw new TypeError('You should enter Array');
+    }
+    if (dice.length !== 5) {
+        throw new TypeError('Need throw 6 dices');
+    }
+    for (i=0 ; i < dice.length; i++) {
+        if (!isInteger(dice[i])) {
+            throw new TypeError('Значения должны быть целые');
Почему у тебя некоторые сообщения на русском, а некоторые на английском?In tests/getPokerHand-test.js:> +        } catch (error) {
+            assert.equal(error.message, 'Need throw 6 dices');
+        }
+    });
+    it('Бросить нужно не более 5', () => {
+        try {
+            getPokerHand([1,2,3,4,4,4,4]);
+            throw new Error('`getPokerHand` should throw error')
+        } catch (error) {
+            assert.equal(error.message, 'Need throw 6 dices');
+        }
+    });
+
+    });
+
+    describe('negative2', () => {    
Почему эти два теста лежат в отдельном describe?Чем они примечательны? Можно ли их опустить на уровень "негативных тестов"?In tests/getPokerHand-test.js:> +            getPokerHand([1, 2, 3, 4, 7]);
+            throw new Error('`getPokerHand` should throw error')
+        } catch (error) {
+            assert.equal(error.message, 'Значения должны быть в промежутке от 1 до 6');
+        }
+    });
+    it('Значения должны быть целые', () => {
+        try {
+            getPokerHand([1.7, 2, 3, 5.4, 5.1]);
+            throw new Error('`getPokerHand` should throw error')
+        } catch (error) {
+            assert.equal(error.message, 'Значения должны быть целые');
+        }
+    });
+
+});
Я обычно оформляю тесты так:describe(`getPokerHand `, () => {
    describe('positive', () => {
        // Здесь собираю тесты на позитивные сценарии
    });
    describe('positive', () => {
        // Здесь собираю негативные тесты
    });
});In tests/getPokerHand-test.js:> +            getPokerHand([1, 2, 3, 4, 7]);
+            throw new Error('`getPokerHand` should throw error')
+        } catch (error) {
+            assert.equal(error.message, 'Значения должны быть в промежутке от 1 до 6');
+        }
+    });
+    it('Значения должны быть целые', () => {
+        try {
+            getPokerHand([1.7, 2, 3, 5.4, 5.1]);
+            throw new Error('`getPokerHand` should throw error')
+        } catch (error) {
+            assert.equal(error.message, 'Значения должны быть целые');
+        }
+    });
+
+});
Старайся сделать так, чтобы в файле был только один describe, всё остальное вкладывай в него.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.    --------------------С уважением, Андрей Комулов 

Andreyproger committed:
Update getPokerHand.js

Changed all file.

Andreyproger committed:
Update getPokerHand-test.js

Changed all structure file

Zhigalov reviewed (approved):
Задание принято!

Andreyproger commented:
Добрый вечер. Сергей, вторая задача по автотестам первый раз была отправлена в срок. В связи с окончанием учебы и написанием диплома не представлялось возможности её поправить. Поправил вместе с первой. Если имеется возможность - прими её пожалуйста.   14.05.2018, 11:29, "Sergey Zhigalov" <notifications@github.com>:Closed #8.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.    --------------------С уважением, Андрей Комулов 
